app/UI/message.ts

```diff
@@ -1,166 +1,106 @@
+    { name: "npub", regex: /(nostr:)?npub[0-9a-z]{59}/ },
+    { name: "nprofile", regex: /(nostr:)?nprofile[0-9a-z]+/ },
+    { name: "naddr", regex: /(nostr:)?naddr[0-9a-z]+/ },
+    { name: "note", regex: /(nostr:)?note[0-9a-z]{59}/ },
+    { name: "nevent", regex: /(nostr:)?nevent[0-9a-z]+/ },
+    { name: "tag", regex: /#\[[0-9]+\]/ },
+];
+
+export function parseContent(content: string): ContentItem[] {
+    if (content.length === 0) {
+        return [];
+    }
+    let max_length_match: {
+        name: ItemType;
+        start: number;
+        end: number;
+    } | undefined;
+    for (const r of regexs) {
+        const mached = r.regex.exec(content);
+        if (mached !== null) {
+            const start = mached.index;
+            const end = mached.index + mached[0].length;
+            // Return the matching string with the maximum length
+            if (!max_length_match || (end - start) > (max_length_match.end - max_length_match.start)) {
+                max_length_match = { name: r.name, start, end };
             }
-        } else {
-            yield {
-                type: type,
-                start: urlStartPosition,
-                end: urlEndPosition,
-            };
         }
     }
+    if (!max_length_match) {
+        return [{ text: content, type: "raw" }];
+    }
+    const text = content.substring(max_length_match.start, max_length_match.end);
+    const bech32 = text.startsWith("nostr:") ? text.slice(6) : text;
+    let matchedItem: ContentItem;
+    if (max_length_match.name === "npub") {
+        const pubkey = PublicKey.FromBech32(bech32);
+        matchedItem = pubkey instanceof Error
+            ? { text, type: "error" }
+            : { text, type: max_length_match.name, pubkey };
+    } else if (max_length_match.name === "nprofile") {
+        const decoded_nProfile = NostrProfile.decode(bech32);
+        matchedItem = decoded_nProfile instanceof Error
+            ? { text, type: "error" }
+            : { text, type: max_length_match.name, pubkey: decoded_nProfile.pubkey };
+    } else if (max_length_match.name === "note") {
+        const noteID = NoteID.FromBech32(bech32);
+        matchedItem = noteID instanceof Error
+            ? { text, type: "error" }
+            : { text, type: max_length_match.name, noteID };
+    } else if (max_length_match.name === "naddr") {
+        const addr = NostrAddress.decode(bech32);
+        matchedItem = addr instanceof Error
+            ? { text, type: "error" }
+            : { text, type: max_length_match.name, addr };
+    } else if (max_length_match.name === "nevent") {
+        const nevent = Nevent.decode(bech32);
+        matchedItem = nevent instanceof Error
+            ? { text, type: "error" }
+            : { text, type: max_length_match.name, nevent };
+    } else {
+        matchedItem = { text, type: max_length_match.name };
+    }
+    return [
+        ...parseContent(content.substring(0, max_length_match.start)),
```

Comment:
Why?

Reply:
Because I didn't add the condition on the far left at the beginning of the match, so there was recursion for this part.